### PR TITLE
Add security extensions

### PR DESCRIPTION
**Pull request by gbiggs**
This PR is a response to [#180](https://github.com/ros2/launch_ros/pull/180). In that PR, a request was made that extensibility be added to the launch system and the `--secure` option be added as an extension. This PR provides the security extensions. The extensibility support is provided in [this PR](https://github.com/ros2/launch_ros/pull/216).

This PR adds:
- An extension to `ros2 launch` that adds the `--secure` and `--no-create-keystore` options.
- An extension to the node action from `launch_ros` that adds the necessary flags to a node's command line to start it with security.

### CODE REVIEW

**Comment by gbiggs** [comment]
Ready for review

**Comment by gbiggs** [comment]
@clalancette Kyle from Canonical is going to take on this package and figure out who to ultimately make maintainer.

File: ros2launch_security/option/security.py
Comment (hidmic) [inline ros2launch_security/option/security.py]: @gbiggs typo:
```suggestion
class NonexistentKeystoreError(Exception):
```
? Or perhaps I'm unaware of this different spelling.

File: test/test_node_action.py
Comment (hidmic) [inline test/test_node_action.py]: @gbiggs unused?

File: test/test_node_action.py
Comment (hidmic) [inline test/test_node_action.py:77]: @gbiggs nit^2: why not inlining `_create node` here? Every other test does so.

File: test/test_option.py
Comment (hidmic) [inline test/test_option.py:41]: @gbiggs invoking `del` on a variable simply decrements its ref count. You may want to force garbage collection for this test to test what you want it to test :) Same elsewhere.

**Review by hidmic — APPROVED** [review]
Overall LGTM pending green CI !

File: ros2launch_security/option/security.py
Comment (gbiggs) [reply ros2launch_security/option/security.py]: Nope, just a typo! Good catch.

File: test/test_node_action.py
Comment (gbiggs) [reply test/test_node_action.py]: Removed in ce8695a.

File: test/test_node_action.py
Comment (gbiggs) [reply test/test_node_action.py:77]: At a guess (I didn't write this code), this is an incomplete refactor.

File: test/test_option.py
Comment (gbiggs) [reply test/test_option.py:41]: Fixed in db0237e.

File: test/test_node_action.py
Comment (gbiggs) [reply test/test_node_action.py:77]: The three occurrences are all slightly different, so I'm inclined to leave these as they are.

File: setup.py
Comment (hidmic) [inline setup.py:20]: @gbiggs not sure if Joe should be a maintainer.

**Review by hidmic — APPROVED** [review]
LGTM but for one comment.

File: setup.py
Comment (gbiggs) [reply setup.py:20]: Good catch! @kyrofa was going to be maintainer of this package, but since he's left Canonical I expect he's no longer available. I don't think there is anyone else lined up yet. I will need to ask the Security WG who can be maintainer before this can be merged.

File: package.xml
Comment (hidmic) [inline package.xml]: @gbiggs we should update this too, if @kyrofa won't be around.

**Comment by hidmic** [comment]
@clalancette how shall we handle running CI for this PR? If I'm not mistaken, this repository won't join the core. A supplemental repos file perhaps?

**Comment by clalancette** [comment]
> @clalancette how shall we handle running CI for this PR? If I'm not mistaken, this repository won't join the core. A supplemental repos file perhaps?

A supplementary repos file would work.  Personally, though, I would just make a primary repos file that contains both this and  the branch for https://github.com/ros2/launch_ros/pull/216 .

**Comment by clalancette** [comment]
Here's CI for this now that ros2/launch_ros#216 is merged:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14237/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9035/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11928/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14341/)

**Comment by clalancette** [comment]
This one has test failures in finding the `sros2.api._keystore` Python module.  It also needs a bit of touch-up according to @hidmic's last review.

**Comment by hidmic** [comment]
I'll take care. Freeze is today and I want to make sure no more changes are needed in the core.

**Comment by hidmic** [comment]
Alright, all issues were in this repository. I've PR'd the changes to this PR in #2. We can run CI once that gets merged into this.

**Comment by hidmic** [comment]
Running CI again:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14323/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9111/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12008/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14427/)

**Comment by hidmic** [comment]
Argh, was missing the `nodl` repo:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14326/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9113/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12012/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14431/)

**Comment by hidmic** [comment]
That should get separately released into Rolling though.

**Comment by hidmic** [comment]
@gbiggs this is good to go. Have you managed to figure out who's going to be the maintainer?

**Comment by gbiggs** [comment]
Not yet. The last security WG meeting was at 2AM. I think I'll send a message instead.

**Comment by kyrofa** [comment]
The new maintainer should be @Arnatious.

**Comment by gbiggs** [comment]
@kyrofa Can you supply an email address for the maintainer line?

File: package.xml
Comment (kyrofa) [inline package.xml]:
```suggestion
  <maintainer email="ted.kern@canonical.com">Ted Kern</maintainer>
```

File: package.xml
Comment (kyrofa) [inline package.xml]:
```suggestion
  <author email="ted.kern@canonical.com">Ted Kern</author>
```

**Comment by gbiggs** [comment]
@hidmic Thanks to @kyrofa we have a maintainer now. The `package.xml` has been updated with their information.

**Comment by Arnatious** [comment]
Just confirming I'm taking over maintaining

**Comment by hidmic** [comment]
@gbiggs @Arnatious what's pending to move forward with this?

**Comment by gbiggs** [comment]
Nothing as far as I know, so I'm going to go ahead and merge it.

**Pull request merged**
